### PR TITLE
Provide an RPC to quit the performance benchmarking workers

### PR DESCRIPTION
**Pull request by vjpai**
I propose adding a "Quit" RPC to the performance benchmarking workers so that it's easier to control them with scripts. Arguably this is not needed.

This will conflict with #4995 which is more important. This is also not complete until this RPC is implemented in the wrapped languages that already support these workers and other language stacks.

@yang-g , is this advisable?

### CODE REVIEW

**Comment by ctiller** [comment]
LGTM

**Comment by vjpai** [comment]
@murgatroid99 Please see the JS code here...

File: src/node/performance/worker_service_impl.js [hunk 1]
```diff
@@ -36,6 +36,11 @@
 var BenchmarkClient = require('./benchmark_client');
 var BenchmarkServer = require('./benchmark_server');
 
+exports.quitWorker = function quitWorker(call, callback) {
+  callback(null, {});
+  process.exit(0);
```
Comment (murgatroid99) [inline src/node/performance/worker_service_impl.js]: This looks basically right, but exiting like that may leave some calls hanging. If that's OK, then this is fine.

Comment (vjpai) [inline src/node/performance/worker_service_impl.js]: So the intention is that this is a separate worker method that gets invoked after the driver is done with the last one. I guess there's an issue here. In C++ we have a lock in the worker such that a worker process can only be running one of the RPCs (RunClient, RunServer, CoreCount, QuitWorker) at a time. Is there an equivalent in Node.js ?

Comment (murgatroid99) [inline src/node/performance/worker_service_impl.js]: The Node server doesn't have anything like that, but I think it's actually OK anyway. Ideally, though, the driver would only call this after other calls to the same worker have completed.

Comment (vjpai) [inline src/node/performance/worker_service_impl.js]: Yes, I think that's the intention. The driver should only invoke this after its previous test has completed so there shouldn't be calls hanging around.

**Pull request merged**